core/whistle_couch-1.0.0/src/couch_mgr.erl

```diff
@@ -528,6 +529,27 @@ cache_db_doc(DbName, DocId, Doc) ->
         {'error', _}=E -> E
     end.
 
+-spec with_cache_doc(text(), ne_binary(), fun((wh_json:object()) -> wh_json:object() | 'skip')) ->
```

Comment:
I can't give you an alternative yet but I don't like this function's name. Will think on this some more, but feel free to offer alternatives that describe the functionality contained.